mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/programmatic/impl/PropertyGenericFieldMappingContextImpl.java

```diff
@@ -33,4 +41,18 @@ PropertyGenericFieldMappingContext thisAsS() {
 		return context;
 	}
 
+	@Override
+	protected <F> void contributeIndexNullAs(StandardIndexFieldTypeContext<?, F> contextConverter, String indexNullAs, Class<?> type) {
+		try {
+			Optional<Object> parsed = ParserUtils.parse( indexNullAs, type );
+			if ( !parsed.isPresent() ) {
+				throw log.unsupportedTypeForIndexNullAs( type );
+			}
+
+			contextConverter.indexNullAs( (F) parsed.get() );
```

Comment:
There are several problems with this solution:

1. It's obviously not extensible. If the user wants to add support for a custom type by writing a custom bridge, he won't be able to use `@GenericField.indexNullAs`.
2. It lacks support for a lot of otherwise supported types. `Instant`, `java.util.Date`, basically all of `java.time`... All those types were supported in Search 5, why wouldn't they be in 6?
3. You ask the user to tell us the expected type, then you cast it to `F` and hope for the best. But we don't have *any* guarantee that `F` is the same type... If the user provides the wrong type, this will fail, and because the cast to `F` is ineffective (generic casts are just cosmetic, they don't do anything), this will fail in a very bizarre way.

So in my opinion, this is not the right approach. 

I think the support for converting from String to `F` should be built in `ValueBridge`. `ValueBridge` is where we know about both the property type and the indexed field type, it's where all the conversion happens and it's what we have to implement to add support for another property type. It feels natural that support for converting from a string to the correct type would be implemented there.

I'll suggest a solution, but some choices were a matter of taste, so feel free to challenge it:

* Expose a `F parse(String value)` method in `ValueBridge`.
* Add the appropriate javadoc to this method.
* Add a default implementation to this method that simply does `throw new UnsupportedOperationException( "Bridge " + toString() + " does not support parsing a value from a String" )`
* Implement this method in all built-in bridges. For the pass-through bridge, I would expect a `Function<String, F>` to be passed to the constructor, and it would be called when `parse` is called.
  Note you could reuse the `ConvertUtils` to generate method references that do what you want : `ConvertUtils::convertBoolean`, `ConvertUtils::convertLong`, etc., and add more methods as necessary. The additional methods may be useful one day for configuration properties.

Now you have extensible support for all property types, but it's not used anywhere. You still need to call this `parse` method:

* Move `org.hibernate.search.engine.mapper.mapping.building.spi.FieldModelContributor` to `org.hibernate.search.mapper.pojo.mapping.building.spi`. I don't know why it was in the engine, it should not be there.
* Change `org.hibernate.search.engine.mapper.mapping.building.spi.FieldModelContributor#contribute` so that it now takes two parameters: `fieldTypeContext` (the current parameter, renamed) and `bridgeContext` (with a new type, `FieldModelContributorBridgeContext`)
* Define `FieldModelContributorBridgeContext` with just one `indexNullAs(String)` method
* Change `org.hibernate.search.mapper.pojo.mapping.definition.programmatic.impl.PojoCompositeFieldModelContributor#contribute` to pass the bridge context to the delegates (which will now be `BiConsumers` instead of just `Consumers`).
* Define an implementation of `FieldModelContributorBridgeContext` that holds a reference to both the `ValueBridge<?, F>` and the `StandardIndexFieldTypeContext<?, F>`. It will be able to implement `indexNullAs(String value)` simply by calling `fieldTypeContext.indexNullAs( bridge.parse( value ) )`.
* Instantiate this implementation in `org.hibernate.search.mapper.pojo.mapping.building.impl.PojoIndexModelBinderImpl#bindValueBridge` add pass it to the field contributor (the line that reads `contributor.contribute( fieldTypeContext );`).